### ISSUE TITLE
Visualization of SSD network

### ISSUE THREAD

**Issue by Bardo91** [yosinski/deep-visualization-toolbox#116]
I am trying to visualize a model trained from https://github.com/weiliu89/caffe/tree/ssd. But I am getting the following error while running the tool:

```
Traceback (most recent call last):
  File "./run_toolbox.py", line 33, in <module>
    main()
  File "./run_toolbox.py", line 23, in main
    lv = LiveVis(settings)
  File "/home/bardo91/programming/3dparty/deep-visualization-toolbox/live_vis.py", line 62, in __init__
    app = app_class(settings, self.bindings)
  File "/home/bardo91/programming/3dparty/deep-visualization-toolbox/caffevis/app.py", line 101, in __init__
    self._populate_net_layer_info()
  File "/home/bardo91/programming/3dparty/deep-visualization-toolbox/caffevis/app.py", line 114, in _populate_net_layer_info
    assert len(blob_shape) in (2,4), 'Expected either 2 for FC or 4 for conv layer'
AssertionError: Expected either 2 for FC or 4 for conv layer
```
Is it some incompatibility of the tool with the network?

These are the settings and the proto model (I added .txt at the end to be able to upload them)
[settings_local.py.txt](https://github.com/yosinski/deep-visualization-toolbox/files/1044503/settings_local.py.txt)
[deploy.prototxt.txt](https://github.com/yosinski/deep-visualization-toolbox/files/1044500/deploy.prototxt.txt)

**Comment by physicso** [comment]
Same error, any solutions?

**Comment by physicso** [comment]
For my case, I have printed out the blob_shape and find that the shape of the last layer is empty. Therefore its length is zero, which leads to this error. I have simply deleted the last loss layer and everything works well :)

**Comment by Bardo91** [comment]
I did some changes and now I am getting  `Check failed: ReadProtoFromBinaryFile(param_file, param) Failed to parse NetParameter file:/home/bardo91/programing/3rdparty/caffe_ssd/use/models/VGGNet/tools465/SSD_300x300/tools465_iter_125000.solverstate ´ any clue?

**Comment by shonehang** [comment]
I am trying to visualize a model trained from SSD too, I meet this mistake:

Cannot copy param 0 weights from layer 'conv4_3_norm_mbox_conf'; shape mismatch. Source param shape is 8 512 3 3 (36864)；target param shape is 12 512 3 3 (55296). To learn this layer's parameters from scratch rather than copying from a saved net, rename the layer.

Now, I solved it, because the deploy.prototxt  I used is not the prototxt I trained before, I change the prototxt and the mistake is solved.
